
-------------- Error (2022-02-28 00:00:07.5090) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`6.Log
         Exception Type: 
         Exception Message: 
         Stack Trace: 
         Additional Info: Failed executing DbCommand (5ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = DateTime2), @p2='?' (DbType = DateTime2), @p3='?' (Size = 4000), @p4='?' (DbType = DateTime2), @p5='?' (Size = 4000), @p6='?' (Size = 4000), @p7='?' (Size = 4000)], CommandType='Text', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Users] ([Guid], [CreatedAt], [DateOfBirth], [Email], [LastUpdated], [Login], [Password], [Role])
VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6, @p7);


-------------- Error (2022-02-28 00:00:07.5301) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`3.Log
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         Additional Info: An exception occurred in the database while saving changes for context type 'PackageTrackingApp.Core.Domain.PackageTrackingContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'Login', table 'PackageTrackingDbV3.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__169_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:d4a121f3-f901-46b3-ba83-27672e5ec092
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)


-------------- Error (2022-02-28 00:00:07.8143) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__9.MoveNext
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at PackageTrackingApp.Infrastructure.Repositories.UserRepository.AddUserAsync(User user) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Repositories\UserRepository.cs:line 24
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 69
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:00:25.1657) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`6.Log
         Exception Type: 
         Exception Message: 
         Stack Trace: 
         Additional Info: Failed executing DbCommand (5ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = DateTime2), @p2='?' (DbType = DateTime2), @p3='?' (Size = 4000), @p4='?' (DbType = DateTime2), @p5='?' (Size = 4000), @p6='?' (Size = 4000), @p7='?' (Size = 4000)], CommandType='Text', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Users] ([Guid], [CreatedAt], [DateOfBirth], [Email], [LastUpdated], [Login], [Password], [Role])
VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6, @p7);


-------------- Error (2022-02-28 00:00:25.1819) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`3.Log
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         Additional Info: An exception occurred in the database while saving changes for context type 'PackageTrackingApp.Core.Domain.PackageTrackingContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'Password', table 'PackageTrackingDbV3.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__169_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:d4a121f3-f901-46b3-ba83-27672e5ec092
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)


-------------- Error (2022-02-28 00:00:25.5094) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__9.MoveNext
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at PackageTrackingApp.Infrastructure.Repositories.UserRepository.AddUserAsync(User user) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Repositories\UserRepository.cs:line 24
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 69
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:00:37.0924) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Please choose a stronger password
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetPassword(String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 72
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 40
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:00:41.3094) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Please choose a stronger password
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetPassword(String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 72
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 40
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:00:44.9038) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Passwords do not match!
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetPassword(String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 77
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 40
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:00:57.1279) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Passwords do not match!
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetPassword(String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 77
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 40
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:01:54.2384) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`6.Log
         Exception Type: 
         Exception Message: 
         Stack Trace: 
         Additional Info: Failed executing DbCommand (7ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = DateTime2), @p2='?' (DbType = DateTime2), @p3='?' (Size = 4000), @p4='?' (DbType = DateTime2), @p5='?' (Size = 4000), @p6='?' (Size = 4000), @p7='?' (Size = 4000)], CommandType='Text', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Users] ([Guid], [CreatedAt], [DateOfBirth], [Email], [LastUpdated], [Login], [Password], [Role])
VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6, @p7);


-------------- Error (2022-02-28 00:01:54.2573) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`3.Log
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         Additional Info: An exception occurred in the database while saving changes for context type 'PackageTrackingApp.Core.Domain.PackageTrackingContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'Email', table 'PackageTrackingDbV3.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__169_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:d4a121f3-f901-46b3-ba83-27672e5ec092
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)


-------------- Error (2022-02-28 00:01:54.6403) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__9.MoveNext
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at PackageTrackingApp.Infrastructure.Repositories.UserRepository.AddUserAsync(User user) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Repositories\UserRepository.cs:line 24
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 69
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:02:01.7058) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`6.Log
         Exception Type: 
         Exception Message: 
         Stack Trace: 
         Additional Info: Failed executing DbCommand (3ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = DateTime2), @p2='?' (DbType = DateTime2), @p3='?' (Size = 4000), @p4='?' (Size = 4000)], CommandType='Text', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Couriers] ([Guid], [DateOfBirth], [DateOfEmployment], [FirstName], [LastName])
VALUES (@p0, @p1, @p2, @p3, @p4);


-------------- Error (2022-02-28 00:02:01.7220) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`3.Log
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         Additional Info: An exception occurred in the database while saving changes for context type 'PackageTrackingApp.Core.Domain.PackageTrackingContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'LastName', table 'PackageTrackingDbV3.dbo.Couriers'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__169_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:d4a121f3-f901-46b3-ba83-27672e5ec092
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)


-------------- Error (2022-02-28 00:02:02.0813) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__9.MoveNext
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at PackageTrackingApp.Infrastructure.Repositories.CourierRepository.AddCourier(Courier courier) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Repositories\CourierRepository.cs:line 23
   at PackageTrackingApp.Infrastructure.Services.CourierService.AddCourierAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\CourierService.cs:line 28
   at PackageTrackingApp.Infrastructure.Handlers.CreateCourierHandler.HandleAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\CreateCourierHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.CourierController.Post(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\CourierController.cs:line 45
   at lambda_method475(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:08:16.6288) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Email string@wp.pl is already taken.
         Stack Trace:    at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 58
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:08:20.4383) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Login string is already taken.
         Stack Trace:    at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 64
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:27:40.4497) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`6.Log
         Exception Type: 
         Exception Message: 
         Stack Trace: 
         Additional Info: Failed executing DbCommand (37ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = Int32), @p2='?' (DbType = Guid), @p3='?' (DbType = Guid), @p4='?' (DbType = DateTime2), @p5='?' (DbType = Single), @p6='?' (DbType = Boolean), @p7='?' (DbType = Boolean), @p8='?' (DbType = Single), @p9='?' (Size = 4000), @p10='?' (DbType = Guid), @p11='?' (DbType = DateTime2), @p12='?' (DbType = Single), @p13='?' (DbType = Single)], CommandType='Text', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Packages] ([Guid], [Category], [CourierGuid], [CustomerGuid], [DeliveredAt], [Height], [IsDelivered], [IsPaid], [Length], [Name], [SellerGuid], [SentAt], [Weight], [Width])
VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6, @p7, @p8, @p9, @p10, @p11, @p12, @p13);


-------------- Error (2022-02-28 00:27:40.5394) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`3.Log
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         Additional Info: An exception occurred in the database while saving changes for context type 'PackageTrackingApp.Core.Domain.PackageTrackingContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): The INSERT statement conflicted with the FOREIGN KEY constraint "FK_Packages_Users_SellerGuid". The conflict occurred in database "PackageTrackingDbV3", table "dbo.Users", column 'Guid'.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__169_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:c7b4753b-9eb4-4c12-809a-226da65bc399
Error Number:547,State:0,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)


-------------- Error (2022-02-28 00:27:40.9145) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__9.MoveNext
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at PackageTrackingApp.Infrastructure.Repositories.PackageRepository.AddAsync(Package package) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Repositories\PackageRepository.cs:line 24
   at PackageTrackingApp.Infrastructure.Services.PackageService.AddAsync(CreatePackage packageDto) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\PackageService.cs:line 36
   at PackageTrackingApp.Infrastructure.Handlers.CreatePackageHandler.HandleAsync(CreatePackage command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\CreatePackageHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.PackageController.Post(CreatePackage package) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\PackageController.cs:line 30
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:35:59.5805) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Email string12@gmail.com is already taken.
         Stack Trace:    at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 58
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:52:16.1308) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Email XDDDDDD@gmail.com is already taken.
         Stack Trace:    at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 58
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:52:22.7676) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: You need to be at least 12 years old to useour services.
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetDateOfBirth(DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 49
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 40
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:52:37.3300) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: You need to be at least 12 years old to useour services.
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetDateOfBirth(DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 49
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 40
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:52:40.7060) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: You need to be at least 12 years old to useour services.
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetDateOfBirth(DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 49
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 40
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:52:45.8573) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: You need to be at least 12 years old to useour services.
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetDateOfBirth(DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 49
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 40
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:52:50.9258) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: You need to be at least 12 years old to useour services.
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetDateOfBirth(DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 49
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 40
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:54:42.0873) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Email XDDDDDDd@gmail.com is already taken.
         Stack Trace:    at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 58
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method438(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:54:48.6514) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`6.Log
         Exception Type: 
         Exception Message: 
         Stack Trace: 
         Additional Info: Failed executing DbCommand (18ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = DateTime2), @p2='?' (Size = 4000), @p3='?' (DbType = DateTime2), @p4='?' (Size = 4000), @p5='?' (Size = 4000), @p6='?' (Size = 4000)], CommandType='Text', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Users] ([Guid], [DateOfBirth], [Email], [LastUpdated], [Login], [Password], [Role])
VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6);


-------------- Error (2022-02-28 00:54:48.6971) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`3.Log
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         Additional Info: An exception occurred in the database while saving changes for context type 'PackageTrackingApp.Core.Domain.PackageTrackingContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'CreatedAt', table 'PackageTrackingDbV3.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__169_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:5af04650-f2ea-4cbd-ad77-50674dd28e7f
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)


-------------- Error (2022-02-28 00:54:49.1002) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__9.MoveNext
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at PackageTrackingApp.Infrastructure.Repositories.UserRepository.AddUserAsync(User user) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Repositories\UserRepository.cs:line 24
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 70
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method438(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:57:36.1105) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Login must be at least 4 characters long
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetLogin(String login) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 79
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 36
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method438(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:57:41.1247) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Email Xddd@gmail.com is already taken.
         Stack Trace:    at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 58
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method438(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:57:45.1239) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`6.Log
         Exception Type: 
         Exception Message: 
         Stack Trace: 
         Additional Info: Failed executing DbCommand (18ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = DateTime2), @p2='?' (Size = 4000), @p3='?' (DbType = DateTime2), @p4='?' (Size = 4000), @p5='?' (Size = 4000), @p6='?' (Size = 4000)], CommandType='Text', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Users] ([Guid], [DateOfBirth], [Email], [LastUpdated], [Login], [Password], [Role])
VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6);


-------------- Error (2022-02-28 00:57:45.1635) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`3.Log
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         Additional Info: An exception occurred in the database while saving changes for context type 'PackageTrackingApp.Core.Domain.PackageTrackingContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'CreatedAt', table 'PackageTrackingDbV3.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__169_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:1c9e4cab-8941-4e8a-ae5e-6f841d68ae53
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)


-------------- Error (2022-02-28 00:57:45.5520) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__9.MoveNext
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at PackageTrackingApp.Infrastructure.Repositories.UserRepository.AddUserAsync(User user) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Repositories\UserRepository.cs:line 24
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 70
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method438(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 00:59:54.7161) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`6.Log
         Exception Type: 
         Exception Message: 
         Stack Trace: 
         Additional Info: Failed executing DbCommand (4ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = DateTime2), @p2='?' (Size = 4000), @p3='?' (DbType = DateTime2), @p4='?' (Size = 4000), @p5='?' (Size = 4000), @p6='?' (Size = 4000)], CommandType='Text', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Users] ([Guid], [DateOfBirth], [Email], [LastUpdated], [Login], [Password], [Role])
VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6);


-------------- Error (2022-02-28 00:59:54.7299) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`3.Log
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         Additional Info: An exception occurred in the database while saving changes for context type 'PackageTrackingApp.Core.Domain.PackageTrackingContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'CreatedAt', table 'PackageTrackingDbV3.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__169_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:1c9e4cab-8941-4e8a-ae5e-6f841d68ae53
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)


-------------- Error (2022-02-28 00:59:55.0231) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__9.MoveNext
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at PackageTrackingApp.Infrastructure.Repositories.UserRepository.AddUserAsync(User user) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Repositories\UserRepository.cs:line 24
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 70
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method438(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:00:46.0945) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`6.Log
         Exception Type: 
         Exception Message: 
         Stack Trace: 
         Additional Info: Failed executing DbCommand (9ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = DateTime2), @p2='?' (Size = 4000), @p3='?' (DbType = DateTime2), @p4='?' (Size = 4000), @p5='?' (Size = 4000), @p6='?' (Size = 4000)], CommandType='Text', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Users] ([Guid], [DateOfBirth], [Email], [LastUpdated], [Login], [Password], [Role])
VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6);


-------------- Error (2022-02-28 01:00:46.1118) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`3.Log
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         Additional Info: An exception occurred in the database while saving changes for context type 'PackageTrackingApp.Core.Domain.PackageTrackingContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'CreatedAt', table 'PackageTrackingDbV3.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__169_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:1c9e4cab-8941-4e8a-ae5e-6f841d68ae53
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)


-------------- Error (2022-02-28 01:00:46.4361) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__9.MoveNext
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at PackageTrackingApp.Infrastructure.Repositories.UserRepository.AddUserAsync(User user) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Repositories\UserRepository.cs:line 24
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 70
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method438(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:04:06.3573) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Email Xdddd@gmail.com is already taken.
         Stack Trace:    at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 58
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method438(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:04:08.2576) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`6.Log
         Exception Type: 
         Exception Message: 
         Stack Trace: 
         Additional Info: Failed executing DbCommand (18ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = DateTime2), @p2='?' (Size = 4000), @p3='?' (DbType = DateTime2), @p4='?' (Size = 4000), @p5='?' (Size = 4000), @p6='?' (Size = 4000)], CommandType='Text', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Users] ([Guid], [DateOfBirth], [Email], [LastUpdated], [Login], [Password], [Role])
VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6);


-------------- Error (2022-02-28 01:04:08.3224) --------------

         Call Site: Microsoft.EntityFrameworkCore.Diagnostics.EventDefinition`3.Log
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         Additional Info: An exception occurred in the database while saving changes for context type 'PackageTrackingApp.Core.Domain.PackageTrackingContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'CreatedAt', table 'PackageTrackingDbV3.dbo.Users'; column does not allow nulls. INSERT fails.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__169_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:a98b9a3b-3e7c-4e86-b4d7-27f92f8d0e10
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)


-------------- Error (2022-02-28 01:04:08.8856) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__9.MoveNext
         Exception Type: Microsoft.EntityFrameworkCore.DbUpdateException
         Exception Message: An error occurred while updating the entries. See the inner exception for details.
         Stack Trace:    at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at PackageTrackingApp.Infrastructure.Repositories.UserRepository.AddUserAsync(User user) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Repositories\UserRepository.cs:line 24
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 70
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method438(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:08:12.0887) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: You need to be at least 12 years old to useour services.
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetDateOfBirth(DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 50
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 41
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:08:23.2091) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: You need to be at least 12 years old to useour services.
         Stack Trace:    at PackageTrackingApp.Core.Domains.User.SetDateOfBirth(DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 50
   at PackageTrackingApp.Core.Domains.User..ctor(String email, String login, String role, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\User.cs:line 41
   at PackageTrackingApp.Infrastructure.Services.UserService.RegisterAsync(String email, String login, String password, String confirmPassword, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\UserService.cs:line 67
   at PackageTrackingApp.Infrastructure.Handlers.RegisterUserHandler.HandleAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\RegisterUserHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.UserController.RegisterAsync(RegisterUser command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\UserController.cs:line 28
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:16:09.6493) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: First name must contain only letters.
         Stack Trace:    at PackageTrackingApp.Core.Domain.Courier.SetFirstName(String firstName) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\Courier.cs:line 45
   at PackageTrackingApp.Core.Domain.Courier..ctor(String firstName, String lastName, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\Courier.cs:line 26
   at lambda_method448(Closure , Object , Courier , ResolutionContext )
   at AutoMapper.Mapper.MapCore[TSource,TDestination](TSource source, TDestination destination, ResolutionContext context, Type sourceType, Type destinationType, IMemberMap memberMap)
   at AutoMapper.Mapper.Map[TSource,TDestination](TSource source, TDestination destination)
   at AutoMapper.Mapper.Map[TDestination](Object source)
   at PackageTrackingApp.Infrastructure.Services.CourierService.AddCourierAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\CourierService.cs:line 27
   at PackageTrackingApp.Infrastructure.Handlers.CreateCourierHandler.HandleAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\CreateCourierHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.CourierController.Post(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\CourierController.cs:line 45
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:16:13.8381) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.ArgumentNullException
         Exception Message: Value cannot be null. (Parameter 'source')
         Stack Trace:    at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.Enumerable.All[TSource](IEnumerable`1 source, Func`2 predicate)
   at PackageTrackingApp.Core.Domain.Courier.SetLastName(String lastName) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\Courier.cs:line 33
   at PackageTrackingApp.Core.Domain.Courier..ctor(String firstName, String lastName, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\Courier.cs:line 27
   at lambda_method448(Closure , Object , Courier , ResolutionContext )
   at AutoMapper.Mapper.MapCore[TSource,TDestination](TSource source, TDestination destination, ResolutionContext context, Type sourceType, Type destinationType, IMemberMap memberMap)
   at AutoMapper.Mapper.Map[TSource,TDestination](TSource source, TDestination destination)
   at AutoMapper.Mapper.Map[TDestination](Object source)
   at PackageTrackingApp.Infrastructure.Services.CourierService.AddCourierAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\CourierService.cs:line 27
   at PackageTrackingApp.Infrastructure.Handlers.CreateCourierHandler.HandleAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\CreateCourierHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.CourierController.Post(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\CourierController.cs:line 45
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:17:05.4094) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.ArgumentNullException
         Exception Message: Value cannot be null. (Parameter 'source')
         Stack Trace:    at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.Enumerable.All[TSource](IEnumerable`1 source, Func`2 predicate)
   at PackageTrackingApp.Core.Domain.Courier.SetLastName(String lastName) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\Courier.cs:line 33
   at PackageTrackingApp.Core.Domain.Courier..ctor(String firstName, String lastName, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\Courier.cs:line 27
   at lambda_method448(Closure , Object , Courier , ResolutionContext )
   at AutoMapper.Mapper.MapCore[TSource,TDestination](TSource source, TDestination destination, ResolutionContext context, Type sourceType, Type destinationType, IMemberMap memberMap)
   at AutoMapper.Mapper.Map[TSource,TDestination](TSource source, TDestination destination)
   at AutoMapper.Mapper.Map[TDestination](Object source)
   at PackageTrackingApp.Infrastructure.Services.CourierService.AddCourierAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\CourierService.cs:line 27
   at PackageTrackingApp.Infrastructure.Handlers.CreateCourierHandler.HandleAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\CreateCourierHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.CourierController.Post(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\CourierController.cs:line 45
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:17:16.1439) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.ArgumentNullException
         Exception Message: Value cannot be null. (Parameter 'source')
         Stack Trace:    at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.Enumerable.All[TSource](IEnumerable`1 source, Func`2 predicate)
   at PackageTrackingApp.Core.Domain.Courier.SetLastName(String lastName) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\Courier.cs:line 33
   at PackageTrackingApp.Core.Domain.Courier..ctor(String firstName, String lastName, DateTime dateOfBirth) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp\Domain\Courier.cs:line 27
   at lambda_method448(Closure , Object , Courier , ResolutionContext )
   at AutoMapper.Mapper.MapCore[TSource,TDestination](TSource source, TDestination destination, ResolutionContext context, Type sourceType, Type destinationType, IMemberMap memberMap)
   at AutoMapper.Mapper.Map[TSource,TDestination](TSource source, TDestination destination)
   at AutoMapper.Mapper.Map[TDestination](Object source)
   at PackageTrackingApp.Infrastructure.Services.CourierService.AddCourierAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\CourierService.cs:line 27
   at PackageTrackingApp.Infrastructure.Handlers.CreateCourierHandler.HandleAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\CreateCourierHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.CourierController.Post(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\CourierController.cs:line 45
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:17:21.1964) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Last name must contain only letters.
         Stack Trace:    at PackageTrackingApp.Core.Domain.Courier.SetLastName(String lastName)
   at PackageTrackingApp.Core.Domain.Courier..ctor(String firstName, String lastName, DateTime dateOfBirth)
   at lambda_method448(Closure , Object , Courier , ResolutionContext )
   at AutoMapper.Mapper.MapCore[TSource,TDestination](TSource source, TDestination destination, ResolutionContext context, Type sourceType, Type destinationType, IMemberMap memberMap)
   at AutoMapper.Mapper.Map[TSource,TDestination](TSource source, TDestination destination)
   at AutoMapper.Mapper.Map[TDestination](Object source)
   at PackageTrackingApp.Infrastructure.Services.CourierService.AddCourierAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\CourierService.cs:line 27
   at PackageTrackingApp.Infrastructure.Handlers.CreateCourierHandler.HandleAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\CreateCourierHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.CourierController.Post(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\CourierController.cs:line 45
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.


-------------- Error (2022-02-28 01:17:28.9445) --------------

         Call Site: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke
         Exception Type: System.Exception
         Exception Message: Courier must be at least 18 years old
         Stack Trace:    at PackageTrackingApp.Core.Domain.Courier.SetDateOfBirth(DateTime dateOfBirth)
   at PackageTrackingApp.Core.Domain.Courier..ctor(String firstName, String lastName, DateTime dateOfBirth)
   at lambda_method448(Closure , Object , Courier , ResolutionContext )
   at AutoMapper.Mapper.MapCore[TSource,TDestination](TSource source, TDestination destination, ResolutionContext context, Type sourceType, Type destinationType, IMemberMap memberMap)
   at AutoMapper.Mapper.Map[TSource,TDestination](TSource source, TDestination destination)
   at AutoMapper.Mapper.Map[TDestination](Object source)
   at PackageTrackingApp.Infrastructure.Services.CourierService.AddCourierAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Services\CourierService.cs:line 27
   at PackageTrackingApp.Infrastructure.Handlers.CreateCourierHandler.HandleAsync(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Handlers\CreateCourierHandler.cs:line 22
   at PackageTrackingApp.Infrastructure.Commands.CommandDispatcher.DispatchAsync[T](T command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Infrastructure\Commands\CommandDispatcher.cs:line 29
   at PackageTrackingApp.Api.Controllers.CourierController.Post(CreateCourier command) in C:\Users\dravi\source\repos\PackageTrackingApp\PackageTrackingApp.Api\Controllers\CourierController.cs:line 45
   at lambda_method443(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
         Additional Info: An unhandled exception has occurred while executing the request.

